File: python/7 kyu/173.py
# https://www.codewars.com/kata/5a005f4fba2a14897f000086/train/python

# Passed

from string import ascii_lowercase

def sum_it_up(numbers_with_bases):
    result = 0
    letters_to_idx_dict = {letter: idx for idx, letter in enumerate(ascii_lowercase)}
    for info in numbers_with_bases:
        number, base = info
        for pow, num in enumerate(number[::-1]):
            if not num.isdigit():
                num = 10 + letters_to_idx_dict[num.lower()]
            result += base ** pow * int(num)
            
    return result

output = sum_it_up([["ABC",16], ["11",2]])
print(output)

File: nodejs/7 kyu/173.js
// https://www.codewars.com/kata/588a3c3ef0fbc9c8e1000095/train/javascript

// Passed

function maxDiff(list) {
  if (list.length === 0 || list.length === 1) return 0;

  let maximum = Math.max(...list);
  let minimum = Math.min(...list);
  let difference = maximum - minimum;
  return difference;
}

const output = maxDiff([0, 1, 2, 3, 4, 5, 16]);
console.log(output);